src/aics-image-viewer/shared/utils/controlPointsToLut.ts

```diff
@@ -126,10 +126,23 @@ export function initializeLut(
 }
 
 export function controlPointsToRamp(controlPoints: ControlPoint[]): [number, number] {
-  if (controlPoints.length === 1 || controlPoints.length === 3) {
+  if (controlPoints.length === 1) {
     return [0, TFEDITOR_MAX_BIN];
   } else if (controlPoints.length === 2) {
     return [controlPoints[0].x, controlPoints[1].x];
+  } else if (controlPoints.length === 3) {
+    if (
+      controlPoints[0].opacity !== controlPoints[1].opacity &&
+      controlPoints[0].opacity !== controlPoints[2].opacity &&
+      controlPoints[1].opacity !== controlPoints[2].opacity
+    ) {
+      // if all 3 are unequal, assume a ramp from first to last
+      return [controlPoints[0].x, controlPoints[2].x];
```

Comment:
Could this be reduced to just a check for `controlPoints[0].opacity !== controlPoints[2].opacity`?

Reply:
if you only check c0 != c2, then c0 could still equal c1, or c1 could be equal to c2.  You still need to make some decision on which are the two best control points to choose.